### DISCUSSION TITLE
bug: Transaction failed: TransactionExecutionError: The total cost (gas * gas fee + value) of executing this transaction exceeds the balance of the account.

### DISCUSSION THREAD

**Discussion by altbee** [wevm/viem#1909] [Question]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.7.19

### Current Behavior

hey, i am trying to swap ETH to USDT 

and it looks like gas calculation is wrong (fyi, i have enough balance on my account)

![image](https://github.com/wevm/viem/assets/34900782/3f0ec4ea-a209-4ff8-b4e9-cd01480cbfdd)

I prepared a simple repo to replicate the issue.

https://github.com/altbee/0x-test

**Comment by joeydefi** [comment]
Stuck with this issue for 1+ week now when using `publicClient.estimateFeesPerGas()` ... anyone has any pointers on how to resolve it? seems like an unit conversion issue somewhere within viem


```js
const { maxFeePerGas, maxPriorityFeePerGas } = await publicClient.estimateFeesPerGas()

const transactionRequest = await client.prepareTransactionRequest({
    account,
    to: to,
    data: data,
    value: value,
    maxFeePerGas: maxFeePerGas,
    maxPriorityFeePerGas: maxPriorityFeePerGas,
})
```


Throws following error even when there's ETH in the wallet...

`     details: 'insufficient funds for gas * price + value: balance 410484309517332303, tx cost 17087486580159439523788579521305283932723, overshot 17087486580159439523788169036995766600420',
      docsPath: undefined,
      metaMessages: [Array],
      shortMessage: 'Missing or invalid parameters.\n' +
        'Double check you have provided the correct parameters.',
      version: 'viem@2.7.19',
      cause: [RpcRequestError],
      code: -32000`

**Reply by joeydefi** [reply]
(happy to fund expedited resolution with $1-2k)

**Reply by jxom** [reply]
Try a different RPC provider maybe

**Reply by jxom** [reply]
Otherwise, please provide a minimal reproducible example.

**Reply by joeydefi** [reply]
Same issue as in here: https://github.com/altbee/0x-test

**Reply by jxom** [reply]
Works fine for me.

**Reply by joeydefi** [reply]
well this is odd but bumping version solved it.. apologies

**Comment by wesleypeeters** [comment]
@altbee Did you ever find a solution to this? I seem to be having a similar issue where the calculation seems to be wrong:

```
Request Arguments:
  from:          0x1fb54Fee8C25a46B4e7a7FBf783c45b08b3583C7
  to:            0xF8D371421790d718B0Fb37F2D3e26d4bD4BB30CE
  value:         0.0123344 ETH
  gas:           25200
  maxFeePerGas:  20 gwei

Details: insufficient funds for gas * price + value: balance 32268726353857000, tx cost 16740563224593921493049097520008881745968, overshot 16740563224593921493049065251282527888968
```
